### ISSUE TITLE
Relative file path with no extension

### ISSUE THREAD

**Issue by KevinSayers** [chapmanb/bcbio-nextgen#1961]
There may be an issue with a relative path with no file extension. Using some of the example files it seems a string as a relative path might be misinterpreted somewhere along the line?

Error:
```

Traceback (most recent call last):
  File "/usr/local/bin/bcbio_nextgen.py", line 219, in <module>
    runfn.process(kwargs["args"])
  File "/usr/local/share/bcbio-nextgen/anaconda/lib/python2.7/site-packages/bcbio/distributed/runfn.py", line 59, in process
    _write_out_argfile(argfile, out, fnargs, parallel, out_keys, work_dir)
  File "/usr/local/share/bcbio-nextgen/anaconda/lib/python2.7/site-packages/bcbio/distributed/runfn.py", line 122, in _write_out_argfile
    recs = [_collapse_to_cwl_record_single(utils.to_single_data(xs)) for xs in out]
  File "/usr/local/share/bcbio-nextgen/anaconda/lib/python2.7/site-packages/bcbio/distributed/runfn.py", line 363, in _collapse_to_cwl_record_single
    out[key] = _to_cwl(tz.get_in(key_parts, data))
  File "/usr/local/share/bcbio-nextgen/anaconda/lib/python2.7/site-packages/bcbio/distributed/runfn.py", line 403, in _to_cwl
    for fname in os.listdir(cur_dir):
OSError: [Errno 2] No such file or directory: ''

```

Causes the error:
```

reference__snap__indexes=Genome 

```

Fixes the error:
```

reference__snap__indexes=./Genome 

```

complete command used:
```

bcbio_nextgen.py runfn alignment_to_rec cwl sentinel_runtime=cores,2,ram,4000 sentinel_parallel=multi-combined sentinel_outputs=alignment_rec files=7_100326_FC6107FAAXX.bam config__algorithm__align_split_size=25000 reference__fasta__base=hg19.fa rgnames__pl=illumina rgnames__sample=Test1 rgnames__pu=Test1 rgnames__lane=Test1 rgnames__rg=Test1  reference__snap__indexes=Genome config__algorithm__aligner=snap config__algorithm__mark_duplicates=True description=Test1

```

**Comment by chapmanb** [comment]
Kevin;
Thanks for the heads up on the problem. I pushed a fix which should hopefully resolve this. Let me know if you get stuck at all.

We've also moved in bcbio to using a JSON based input instead of trying to serialize on the command line like this. The recent bcbio development version should use this approach, which is hopefully easier to manage since bcbio handles the serialization from the inputs instead of relying on specific command lines. Hope this helps and thanks again for the report.
